kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -51,50 +52,75 @@
 log = logging.getLogger('collector')
 
 
-class KubeletCheck(AgentCheck, CadvisorScraper):
+class KubeletCheck(CadvisorPrometheusScraperMixin, GenericPrometheusCheck, CadvisorScraper):
     """
     Collect metrics from Kubelet.
     """
 
     def __init__(self, name, init_config, agentConfig, instances=None):
-        super(KubeletCheck, self).__init__(name, init_config, agentConfig, instances)
-
         self.NAMESPACE = 'kubernetes'
 
+        super(KubeletCheck, self).__init__(name, init_config, agentConfig, None)
+
         if instances is not None and len(instances) > 1:
             raise Exception('Kubelet check only supports one configured instance.')
         inst = instances[0] if instances else None
 
         self.cadvisor_legacy_port = inst.get('cadvisor_port', CADVISOR_DEFAULT_PORT)
         self.cadvisor_legacy_url = None
 
-        self.cadvisor_scraper = CadvisorPrometheusScraper(self)
-
-        self.kubelet_scraper = PrometheusScraper(self)
-        self.kubelet_scraper.NAMESPACE = 'kubernetes'
-        self.kubelet_scraper.metrics_mapper = {
-            'apiserver_client_certificate_expiration_seconds': 'apiserver.certificate.expiration',
-            'rest_client_requests_total': 'rest.client.requests',
-            'kubelet_runtime_operations': 'kubelet.runtime.operations',
-            'kubelet_runtime_operations_errors': 'kubelet.runtime.errors',
+        self.cadvisor_scraper_config = self._create_cadvisor_prometheus_scraper(inst)
+        self.kubelet_scraper_config = self._create_kubelet_prometheus_scraper(inst)
+
+        self.METRIC_TRANSFORMERS = {
```

Comment:
Would it be more clear to define `METRIC_TRANSFORMERS` in `CadvisorPrometheusScraperMixin` since that's where all the functions are defined?

Reply:
I think both the suggestions sound good!